Find a functionally equivalent code:android.text.Layout.isRtlCharAt:COMMENT
Method Modifier: public      
Comment:/**
 * Returns true if the character at offset is right to left (RTL).
 * @param offset the offset
 * @return true if the character is RTL, false if it is LTR
 */

@@ -10,18 +10,18 @@
     int[] runs = dirs.mDirections;
     int lineStart = getLineStart(line);
     for (int i = 0; i < runs.length; i += 2) {
-        int start = lineStart + (runs[i] & RUN_LENGTH_MASK);
-        // corresponding of the last run
-        if (offset >= start) {
+        int start = lineStart + runs[i];
+        int limit = start + (runs[i + 1] & RUN_LENGTH_MASK);
+        if (offset >= start && offset < limit) {
             int level = (runs[i + 1] >>> RUN_LEVEL_SHIFT) & RUN_LEVEL_MASK;
             return ((level & 1) != 0);
         }

